src/Sentry/Protocol/IHasExtra.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+
+namespace Sentry.Protocol
+{
+    /// <summary>
+    /// Implemented by objects that contain a map of untyped extra data.
+    /// </summary>
+    public interface IHasExtra
```

Comment:
Separating these into these micro interfaces allows us to reuse the same extension methods across Event, Span, Transaction, and Scope.

Reply:
Also note: the `extra` on a span is actually serialized as `data` in JSON (as required by protocol).